File: src/python/compress_ob.py
import re

compress = lambda input: "".join(sum([[str(len(v)), v[0]] for i, v in enumerate(re.split(r'(?<=(.))(?!\1|$)', input)) if i % 2 == 0], []))

assert compress("AAABBAAC") == "3A2B2A1C"

File: src/javascript/compress_ob.js
const assert = require('assert');

const sequence = (string) => 
  string.split(/(?<=(.))(?!\1|$)/g)
    .filter((_, i) => !(i % 2))
    .map((c) => [c.length, c[0]])
    .flatMap((c) => c)
    .join('')

assert.equal(
  sequence("AAABBAAC"),
  "3A2B2A1C"
)
